Include/interactivity.h

```diff
@@ -1,1152 +1,1169 @@
+/// </summary>
+class interactive_control
+{
+public:
+
+    /// <summary>
+    /// The type of control.
+    /// </summary>
+    _MIXERIMP const interactive_control_type& control_type() const;
+
+    /// <summary>
+    /// Unique string identifier for the control.
+    /// </summary>
+    _MIXERIMP const string_t& control_id() const;
+
+protected:
+
+    /// <summary>
+    /// Internal constructor for the interactive_control object.
+    /// </summary>
+    interactive_control();
+
+    /// <summary>
+    /// Internal virtual destructor for the interactive_control object.
+    /// </summary>
+    virtual ~interactive_control()
+    {
+    }
+
+    /// <summary>
+    /// Internal constructor for the interactive_control object.
+    /// </summary>
+    interactive_control(
+        _In_ string_t parentScene,
+        _In_ string_t controlId,
+        _In_ bool disabled
+        );
+
+    /// <summary>
+    /// Internal function to clear the state of the interactive_control object.
+    /// </summary>
+    virtual void clear_state() = 0;
+
+    /// <summary>
+    /// Internal function to update the state of the interactive_control object.
+    /// </summary>
+    virtual bool update(web::json::value json, bool overwrite) = 0;
+
+    /// <summary>
+    /// Internal function to initialize interactive_control object.
+    /// </summary>
+    virtual bool init_from_json(_In_ web::json::value json) = 0;
+
+    std::shared_ptr<interactivity_manager> m_interactivityManager;
+    string_t m_parentScene;
+    interactive_control_type m_type;
+    string_t m_controlId;
+    bool m_disabled;
+    string_t m_etag;
+
+    friend interactive_control_builder;
+    friend interactivity_manager_impl;
+};
+
+
+/// <summary>
+/// Represents an interactivity button control. This class is 
+/// derived from interactive_control. All controls are created and 
+/// configured using the Mixer Interactive Lab.
+/// </summary>
+class interactive_button_control : public interactive_control
+{
+public:
+
+    /// <summary>
+    /// Text displayed on the button control.
+    /// </summary>
+    _MIXERIMP const string_t& button_text() const;
+
+    /// <summary>
+    /// Spark cost assigned to the button control.
+    /// </summary>
+    _MIXERIMP uint32_t cost() const;
+
+    /// <summary>
+    /// Indicates whether the button is enabled or disabled. If TRUE, 
+    /// button is disabled.
+    /// </summary>
+    _MIXERIMP bool disabled() const;
+
+    /// <summary>
+    /// Function to enable or disable the button.
+    /// </summary>
+    /// <param name="disabled">Value to enable or disable the button. 
+    /// Set this value to TRUE to disable the button.</param>
+    _MIXERIMP void set_disabled(_In_ bool disabled);
+
+    /// <summary>
+    /// Sets the progress value for the button control.
+    /// </summary>
+    /// <param name="progress">The progress value, in the range of 0.0 to 1.0.</param>
+    _MIXERIMP void set_progress(_In_ float progress);

```

Comment:
sounds good. I'll put it back to where it was. I didn't mean to move it. Thank you for calling that out,